### PR TITLE
Cleaned up Server class

### PR DESCRIPTION
**Pull request by huwpascoe**
In a step towards network data compression, server class needed some cleaning since it hasn't been touched since forever.

Now organized with regions. All those many vague DispatchOrders have been reduced to one method so it's a lot clearer. Added readonly to appropriate fields, removed public access to others, fixed some style issues. Removed the "volatile" field that isn't necessary for an enum (which is an atomic type...and said field is not even set from the server thread)

### CODE REVIEW

**Comment by huwpascoe** [comment]
Rebased taking into account recent commits.

File: OpenRA.Game/Server/Server.cs [hunk 1]
```diff
@@ -194,13 +134,132 @@ public Server(IPEndPoint endpoint, ServerSettings settings, ModData modData)
...
+			foreach (var t in serverTraits.WithInterface<IStartGame>())
+				t.GameStarted(this);
+
+			// Check TimeOut
+			if (Settings.TimeOut > 10000)
+			{
+				var gameTimeout = new Timer(state =>
+				{
+					((Timer)state).Dispose();
+					Console.WriteLine("Timeout at {0}!!!", Settings.TimeOut);
```
Comment (Phrohdoh) [inline OpenRA.Game/Server/Server.cs]: No need for `!!!`.

File: OpenRA.Game/Server/Server.cs [hunk 2]
```diff
@@ -578,11 +509,120 @@ public void DropClient(Connection toDrop, int frame)
...
+		}
+
+		#endregion
+
+		#region Sync Info
+
+		public static void SyncClientToPlayerReference(Session.Client c, PlayerReference pr)
+		{
+			if (pr == null)
+				return;
```
Comment (Phrohdoh) [inline OpenRA.Game/Server/Server.cs]: Need a blank line after each instruction (following a condition):
```csharp
if (foo)
    return;

if (wow)
..
```

**Comment by huwpascoe** [comment]
Fixed ye olde styling

File: OpenRA.Game/Server/Server.cs (hunk 2 above)
Comment (huwpascoe) [inline OpenRA.Game/Server/Server.cs]: not sure why this note doesn't go away, updated three times since it was posted

Comment (cjshmyr) [inline OpenRA.Game/Server/Server.cs]: It was a comment on line 609 which hasn't changed

**Comment by huwpascoe** [comment]
Up to date, need review/merge

**Comment by Mailaender** [comment]
The multiplayer lobby still seems to work. This is very hard to review because of the copy and paste of larger code blocks.

**Comment by huwpascoe** [comment]
Hm.. I hadn't thought of that. Okay, I'll split this PR up.

**Pull request closed without merging**